pkg/gadgets/trace/byob/tracer/tracer.go

```diff
@@ -0,0 +1,346 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tracer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/btf"
+	"github.com/cilium/ebpf/link"
+	"github.com/cilium/ebpf/perf"
+	"github.com/cilium/ebpf/ringbuf"
+	"github.com/solo-io/bumblebee/pkg/decoder"
+	beespec "github.com/solo-io/bumblebee/pkg/spec"
+	orascontent "oras.land/oras-go/pkg/content"
+	"oras.land/oras-go/pkg/oras"
+
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/byob/types"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+const (
+	BPFSocketAttach = 50
+)
+
+type Config struct {
+	RegistryAuth orascontent.RegistryOptions
+	ProgLocation string
+	ProgContent  []byte
+	MountnsMap   *ebpf.Map
+}
+
+type Tracer struct {
+	config         *Config
+	enricher       gadgets.DataEnricherByMntNs
```

Comment:
It's not required after https://github.com/inspektor-gadget/inspektor-gadget/pull/1281. The operator will enrich the data automatically.